source/CIS/cis-2.rst

```diff
@@ -0,0 +1,790 @@
+The reason for not also including a single transfer function is to have smaller smart contract modules, which in turn leads to saving cost on every function call.
+
+No explicit authorization
+-------------------------
+
+The specification does not mandate any authorization scheme and one might expect a requirement for the owner and operators being authorized to transfer tokens.
+This is very much intentional and the reasoning for this is to keep the specification focused on the interface for transferring token ownership with as few restrictions as possible.
+
+Having a requirement that only owners and operators can transfer would prevent introducing any other authorization scheme on top of this specification.
+
+Adding a requirement for owners and operators being authorized to transfer tokens would prevent introducing custom contract logic rejecting transfers, such as limiting the daily transfers, temporary token lockups or non-transferrable tokens.
+
+Instead, this specification includes a requirement to ensure transfers by operators are executed as if they are sent by the owner, meaning whenever a token owner is authorized, so is an operator of the owner.
+
+Most contracts implementing this specification should probably add some authorization and not have anyone being able to transfer any token, but this is not in the scope of this standard.
+
+No token-level approval/allowance like in ERC20 and ERC721
+----------------------------------------------------------
+
+This standard only specifies address-level operators and not token-level operators.
+The main argument is simplicity and to save energy cost on common cases, but other reasons are:
+
+- Token-level operators require the token smart contract to track more state, which increases the overall energy cost.
+- For token smart contracts with a lot of token types, such as a smart contract with a large collection of NFTs, token-level operators could become very expensive.
+- For fungible tokens; `approval/allowance introduces an attack vector <https://docs.google.com/document/d/1YLPtQxZu1UAvO9cZ1O2RPXBbT0mooh4DYKjA_jp-RLM/edit>`_.
+
+.. note::
+
+  The specification does not prevent adding more fine-grained authorization, such as token-level operators.
+
+Receive hook function
+---------------------
+
+The specification requires a token receive hook to be called on a smart contract receiving tokens, this will in some cases prevent mistakes such as sending tokens to smart contracts which do not define behavior for receiving tokens.
+These token could then be lost forever.
+
+The reason for this not being optional is to allow other smart contracts which integrate with a token smart contract to rely on this for functionality.
+
+.. warning::
+
+  The smart contract receive hook function could be called by any smart contract and it is up to the integrating contract whether to trust the token contract.
```

Comment:
It could be any caller, not just a token contract.
```suggestion
  The smart contract receive hook function could be called by any smart contract and it is up to the integrating contract whether to trust the caller.
```